src/tests/test_output.py

```diff
@@ -0,0 +1,43 @@
+import os
+import argparse
+import subprocess
+import pytest
+
+
+def extract_files_with_extension(dir_name, extension):
+    return [name for name in os.listdir(dir_name) if name.endswith(extension)]
+
+
+cur_dir = os.path.dirname(os.path.realpath(__file__))
+TEST_FILES = [os.path.splitext(name)[0] for name in
+                extract_files_with_extension(cur_dir, '.c')]
+
+
+def setup_module(module):
+    subprocess.run('cd {:} && make clean && make'.format(cur_dir), shell=True,
+                   stdout=subprocess.PIPE)
+
+
+@pytest.mark.parametrize('name', TEST_FILES)
+def test_output(name):
+    bin_path = '{test_dir}/{name}'.format(test_dir=cur_dir, name=name)
+    p = subprocess.run(bin_path, shell=True, stdout=subprocess.PIPE,
```

Comment:
You are not using `stderr`. Did you want to print it in your error message? If not you can omit `stderr=`

Reply:
Good point!